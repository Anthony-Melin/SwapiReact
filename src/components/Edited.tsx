import LabelDateFactory from './__components__/LabelDateFactory.tsx'

export default LabelDateFactory('edited', 'Edited')
