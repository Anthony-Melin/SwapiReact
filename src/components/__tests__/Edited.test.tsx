import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Edited from '../Edited.tsx'

describe('Edited', () => {
    beforeEach(() => {
        render(<Edited value="2014-12-09T23:50:51.644000Z" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('edited')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('edited')).toHaveTextContent(
            'Edited: 10/12/2014 00:50:51'
        )
    })
})
