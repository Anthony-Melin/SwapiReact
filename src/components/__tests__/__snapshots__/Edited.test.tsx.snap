// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Edited > should match snapshot 1`] = `
<p
  data-testid="edited"
>
  Edited
  : 
  10/12/2014 00:50:51
</p>
`;
